js/apps/admin-ui/src/identity-providers/add/AdvancedSettings.tsx

```diff
@@ -125,6 +136,87 @@ export const AdvancedSettings = ({ isOIDC, isSAML }: AdvancedSettingsProps) => {
       />
       <SwitchField field="config.hideOnLoginPage" label="hideOnLoginPage" />
 
+      <FormGroupField label="filteredByClaim">
+        <Controller
+          name="config.filteredByClaim"
+          defaultValue="false"
+          control={control}
+          render={({ field }) => (
+            <Switch
+              id="filteredByClaim"
+              label={t("common:on")}
+              labelOff={t("common:off")}
+              isChecked={field.value === 'true'}
+              onChange={(value) => {
+                  const v=""+value
+                  field.onChange(v)
+                  setFilteredByClaim(v)
+                  setClaimFilterRequired(value)
+                  console.debug('setFilteredByClaim ' + v)
+                }
+              }
+              isDisabled={false}
+              aria-label="filteredByClaim"
+            />
+          )}
+        />
+      </FormGroupField>
+      { claimFilterRequired && (
+        <>
+          <FormGroup
+            label={t("identity-providers:claimFilterName")}
+            labelIcon={
+              <HelpItem
+                helpText={t("identity-providers-help:claimFilterName")}
+                fieldLabelId="identity-providers:claimFilterName"
+              />
+            }
+            fieldId="kc-claim-filter-name"
+            isRequired={claimFilterRequired}
+            validated={
+              errors.config?.claimFilterName
+                ? ValidatedOptions.error
+                : ValidatedOptions.default
+            }
+            helperTextInvalid={t("common:required")}
+          >
+            <KeycloakTextInput
+              isRequired={claimFilterRequired}
```

Comment:
because `claimFilterRequired` is always true this can be shorted:
```suggestion
              isRequired
```

Reply:
thanks